source/game.ai.bmx

```diff
@@ -480,6 +491,17 @@ Type TLuaFunctions Extends TLuaFunctionsBase {_exposeToLua}
 	End Method
 
 
+	'send figure to elevator (plan) on the current floor
+	Method doGoToElevatorPlan:Int()
```

Comment:
Ich weiß nicht, ob das mit der aktuellen Engine zusammenpasst, da der GoToRoom-Job eine ID haben möchte, die er dann ansteuert.
Im Idealfall würde ich im neu zu erstellenden ElevatorTask in der Activate-Methode einfach "TargetRoom=TVT.GetElevatorOnCurrentFloor()" setzen und der Rest der API bleibt unverändert (oder getFirstRoomByDetails übernimmt das Finden der ID, wenn der Raumname dem Aufzug entspricht.

Mit dieser Methode müsste ich vermutlich die getGoToJob in der Engine überschreiben und den Job (korrekt) implementieren.

Reply:
Die Fahrstuhlplaene sind alles derselbe Raum ... hier betritt man "Hotspots".

Ich weiss nicht mehr genau, warum ich das ueber "THotspot" und nicht ueber "TRoomBase" umgesetzt hatte.

Allerdings gibt es ja auch den "Getraenkeautomaten" oder spaeter vlt noch andere "hotspots" im Hochhaus (das Hochhaus ist auch ein "Raum"). Vlt waere es entsprechend praktischer, ein "doGoToHotspot" einzufuehren - und dann aehnlich der Raeume ein "GetFirstByDetails()" hinzuzufuegen

Reply:
```XML
				<!--
					elevator planner
					================
					instead of rooms in each floor, we create hotspots
					so the building can handle clicks to "other floor's planner"
					without the figure targeting other floors
				-->
```
Vielleicht... ein Hinweis, der beim Abaendern zu beachten waere.

Reply:
Könntest Du die spezifische doGoToElevatorPlan-Methode wieder entfernen?

Reply:
Ich meinte tatsächlich die gesamte Methode in game.ai. Wenn in der Engine Target und Raum definiert werden können, um nicht den GoTo-Job überschreiben zu müssen, dann ist diese API überflüssig.